### ISSUE TITLE
Restart tailer when the process hasn't reported exit status

### ISSUE THREAD

**Issue by dblock** [dblock/heroku-commander#3]
This tailer gave up and exited, but the remote process has not finished. It seems pretty common for long running tasks. Restart it.

```
03:53:16 [2013-02-06 08:53:11 +0000] done.
03:53:16 [2013-02-06 08:53:11 +0000] Rolling up mass ...
03:53:16 Running: heroku run:detached "(rake db:rollup 2>&1 ; echo rc=\$?)" --app mass-production
03:53:23 Started: 4864
03:53:23 4864: Running `(rake db:rollup 2>&1 ; echo rc=$?)` detached... up, run.9226
03:53:23 Heroku pid run.9226 up.
03:53:23 Running: heroku logs -p run.9226 --tail --app mass-production
05:11:42 rake aborted!
05:11:42 
05:11:42 Problem:
05:11:42   The command `rake db:rollup` (pid: run.9226) failed without reporting an exit status.
05:11:42 Summary:
05:11:42   The command rake db:rollup failed.
05:11:42    DEPRECATION WARNING: You have Rails 2.3-style plugins in vendor/plugins! Support for these plugins will be removed in Rails 4.0. Move them out and bundle them in your Gemfile, or fold them in to your app as lib/myplugin/* and config/initializers/myplugin.rb. See the release notes for more on this: http://weblog.rubyonrails.org/2012/1/4/rails-3-2-0-rc2-has-been-released. (called from <top (required)> at /app/Rakefile:5)
05:11:42    DEPRECATION WARNING: You have Rails 2.3-style plugins in vendor/plugins! Support for these plugins will be removed in Rails 4.0. Move them out and bundle them in your Gemfile, or fold them in to your app as lib/myplugin/* and config/initializers/myplugin.rb. See the release notes for more on this: http://weblog.rubyonrails.org/2012/1/4/rails-3-2-0-rc2-has-been-released. (called from <top (required)> at /app/Rakefile:5)
05:11:42    ** [NewRelic][02/06/13 08:53:41 +0000 8f45b328-0221-4f38-bdd7-ef1b2a13b43f (4)] INFO : Starting the New Relic Agent.
05:11:42    ... skipping 22 line(s) ...
05:11:42    [2013-02-06 09:06:14 +0000] Rolling up ChangelogGroupByAssociation ...
05:11:42    [2013-02-06 09:11:41 +0000] Rolling up PartnerStatSnapshot ...
05:11:42 Resolution:
05:11:42   Examine the command line and refer to the command documentation.
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/bundler/gems/heroku-commander-6d3d852fb628/lib/heroku/runner.rb:73:in `check_exit_status!'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/bundler/gems/heroku-commander-6d3d852fb628/lib/heroku/runner.rb:55:in `run_detached!'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/bundler/gems/heroku-commander-6d3d852fb628/lib/heroku/runner.rb:20:in `run!'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/bundler/gems/heroku-commander-6d3d852fb628/lib/heroku/commander.rb:38:in `run'
05:11:42 /mnt/jenkins/workspace/gravity-daily-cron/lib/tasks/heroku.rake:18:in `execute'
05:11:42 /mnt/jenkins/workspace/gravity-daily-cron/lib/tasks/mass_rollup.rake:21:in `block (2 levels) in <top (required)>'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:205:in `call'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:205:in `block in execute'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:200:in `each'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:200:in `execute'
05:11:42 /mnt/jenkins/workspace/gravity-daily-cron/lib/tasks/mass_rollup.rake:6:in `block (3 levels) in <top (required)>'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:205:in `call'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:205:in `block in execute'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:200:in `each'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:200:in `execute'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:158:in `block in invoke_with_call_chain'
05:11:42 /home/jenkins/.rvm/rubies/ruby-1.9.3-p286/lib/ruby/1.9.1/monitor.rb:211:in `mon_synchronize'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:151:in `invoke_with_call_chain'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:144:in `invoke'
05:11:42 /mnt/jenkins/workspace/gravity-daily-cron/lib/tasks/cron.rake:8:in `block (3 levels) in <top (required)>'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:205:in `call'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:205:in `block in execute'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:200:in `each'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:200:in `execute'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:158:in `block in invoke_with_call_chain'
05:11:42 /home/jenkins/.rvm/rubies/ruby-1.9.3-p286/lib/ruby/1.9.1/monitor.rb:211:in `mon_synchronize'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:151:in `invoke_with_call_chain'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/task.rb:144:in `invoke'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/application.rb:112:in `invoke_task'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/application.rb:90:in `block (2 levels) in top_level'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/application.rb:90:in `each'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/application.rb:90:in `block in top_level'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/application.rb:129:in `standard_exception_handling'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/application.rb:84:in `top_level'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/application.rb:62:in `block in run'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/application.rb:129:in `standard_exception_handling'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/lib/rake/application.rb:59:in `run'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/gems/rake-0.9.2/bin/rake:32:in `<top (required)>'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/bin/rake:23:in `load'
05:11:42 /home/jenkins/.rvm/gems/ruby-1.9.3-p286/bin/rake:23:in `<main>'
05:11:42 Tasks: TOP => mass:production:rollup
```

**Comment by fancyremarker** [comment]
I think this is closed by 91b53451e91be68eba8da3e129acdcdc5e6e77ba?

**Comment by dblock** [comment]
Yep.
